app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: false
+
+module Users
+  # Controller for omniauth to get data from outside services
+  class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    # rubocop:disable Metrics/AbcSize
+    def google_oauth2
+      @user = User.from_omniauth(request.env['omniauth.auth'])
+      if @user.persisted?
+        sign_in(@user)
+        current_user.phone_number.nil? ? (redirect_to after_signup_index_path) : (redirect_to root_path)
```

Comment:
```suggestion
        redirect_to current_user.phone_number.nil? ?  after_signup_index_path : root_path
```

check if this solution is workinng